### PR TITLE
Done mongo-fying results page

### PR DESCRIPTION
**Pull request by h94h12**

### CODE REVIEW

**Comment by sergeyk** [comment]
Awesome, I will review this later today. Could we meet tomorrow at 3 or
5:30pm?

**Comment by h94h12** [comment]
I can meet tomorrow at 3 but I have to go by 4.

File: ui/app.py [hunk 1]
```diff
@@ -21,33 +21,49 @@
...
+        for setting in settings:
+            df_ = preds_panel.minor_xs(setting)
+            for style_name in style_names:
+                df_['pred_' + style_name] -= threshold_df.loc[style_name, setting]
+                df_['abs_pred_' + style_name] = np.abs(df_['pred_' + style_name])
+            df_['setting'] = setting
+            df_ = df_.reset_index()
+            insert_df(df_, collection)
+
+settings = []
```
Comment (sergeyk) [inline ui/app.py:67]: Why is this necessary? And why is this passed into load_pred_results?

File: ui/app.py [hunk 2]
```diff
@@ -21,33 +21,49 @@
...
-    # TODO: load into mongo collections instead
-    for setting in preds_panel.minor_axis:
-        df_ = preds_panel.minor_xs(setting)
-        for style_name in style_names:
-            df_['pred_' + style_name] -= threshold_df.loc[style_name, setting]
-            df_['abs_pred_' + style_name] = np.abs(df_['pred_' + style_name])
-        all_preds[experiment_name][setting] = df_
-
-
+    settings += preds_panel.minor_axis
```
Comment (sergeyk) [inline ui/app.py:53]: This gives me an error: `ValueError: operands could not be broadcast together with shapes (0,) (3,)`

**Comment by sergeyk** [comment]
Inserting all the records into DB is quite slow. Experiment with [bulk inserts](http://docs.mongodb.org/manual/core/bulk-inserts/) -- it can probably be made a _lot_ faster

File: ui/app.py [hunk 3]
```diff
@@ -21,33 +21,49 @@
...
     """
     Load prediction results into accessible form.
     Also, make positive predictions positive and negative negative, to
     simplify future sorting by confidence.
     """
-    threshold_df = pd.read_hdf(
-        '{}/{}_thresholds.h5'.format(results_dirname, experiment_name), 'df')
-
+    collection = mongo_client[db_name][experiment_name]
     _, preds_panel = vislab._results.load_pred_results(
```
Comment (sergeyk) [inline ui/app.py:50]: This is still loaded every time, independent of whether any records are in the collection. Should move to inside of the following if block

**Comment by sergeyk** [comment]
Please confirm that it all works correctly and loads quickly before updating PR

File: ui/app.py (hunk 1 above)
Comment (h94h12) [inline ui/app.py:67]: You need to know the list of settings for the select options.

File: ui/app.py (hunk 2 above)
Comment (h94h12) [inline ui/app.py:53]: Huh I am not getting that...

**Comment by sergeyk** [comment]
Baller-town
